### PR TITLE
Fix GHA + support Python 3.11.

### PR DESCRIPTION
**Pull request by icemac**
I had to disable flake8 checks, so isort-ing the imports is not enforced.

### CODE REVIEW

**Comment by mgedmin** [comment]
Some builds are failing with

```
py37-zodbmaster: install_deps> python -I -m pip install -e git+https://github.com/zopefoundation/ZODB.git@master
py37-zodbmaster: exit 1 (1.21 seconds) /home/runner/work/ZEO/ZEO> python -I -m pip install -e git+https://github.com/zopefoundation/ZODB.git@master pid=1656
ERROR: Could not detect requirement name for 'git+https://github.com/zopefoundation/ZODB.git@master', please specify one with #egg=your_package_name
```

presumably a newer version of pip is pickier about the arguments it accepts.

Adding `#egg=ZODB` ought to work here.

**Comment by mgedmin** [comment]
This works for me with tox 4.0.2:

```diff
diff --git a/tox.ini b/tox.ini
index dfa36851..5d077daa 100644
--- a/tox.ini
+++ b/tox.ini
@@ -20,7 +20,7 @@ envlist =
 usedevelop = true
 deps =
     !zodbmaster: ZODB >= 4.2.0b1
-    zodbmaster: -e git+https://github.com/zopefoundation/ZODB.git@master#egg=ZODB
+    zodbmaster: -e "git+https://github.com/zopefoundation/ZODB.git@master\#egg=ZODB"
     uvloop: uvloop
 setenv =
     !py27-!pypy: PYTHONWARNINGS=ignore::ResourceWarning
@@ -63,7 +63,7 @@ deps =
     coverage
     coverage-python-version
     !zodbmaster: ZODB >= 4.2.0b1
-    zodbmaster: -e git+https://github.com/zopefoundation/ZODB.git@master#egg=ZODB
+    zodbmaster: -e "git+https://github.com/zopefoundation/ZODB.git@master\#egg=ZODB"
     uvloop: uvloop
 commands =
     mkdir -p {toxinidir}/parts/htmlcov

```

Just the quotes were insufficient.  I haven't tried just the backslash.

File: .meta.toml [hunk 1]
```diff
@@ -11,9 +11,10 @@ with-future-python = false
 with-legacy-python = true
 with-docs = true
 with-sphinx-doctests = false
+with-macos = false
 
 [tox]
-use-flake8 = true
+use-flake8 = false
```
[the remark this answers is not in the document]
Comment (icemac) [reply .meta.toml:17]: Now it requires imports to be sorted using isort. There are many open PRs which might get conflicts if we do this.

On the other hand I have no problem trying it – any suggestions?

**Comment by icemac** [comment]
> Just the quotes were insufficient. I haven't tried just the backslash. The quotes add a nice safety-check where if you forget the backslash, you get an error ("No closing quotation") instead of silent truncation.

According to https://github.com/tox-dev/tox/issues/2638 just the backslash should be sufficient. I'll add it and switch back to a current tox version.

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2022-12-8 23:01 -0800:
>@icemac commented on this pull request.
> ...
>Now it requires imports to be sorted using isort. There are many open PRs which might get conflicts if we do this.
>
>On the other hand I have no problem trying it – any suggestions?


Is there a chance just to disable the `isort` check
(maybe via a configuration or a parameter)?

It would be a pain to resolve merge conflicts caused by different
sorting rules.

File: .meta.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dataflake) [reply .meta.toml:17]: There is at least one very large PR open from @d-maurer, please hold off with any large "cosmetic" changes until then.

**Comment by icemac** [comment]
> Is there a chance just to disable the `isort` check (maybe via a configuration or a parameter)? It would be a pain to resolve merge conflicts caused by different sorting rules.

Currently there is no such switch. But on the other hand there are currently no enforced sorting rules for this repository, so the possible conflicts have to be resolved only once.

**Comment by dataflake** [comment]
You can always disable it manually in tox.ini.

**Comment by dataflake** [comment]
> > Just the quotes were insufficient. I haven't tried just the backslash. The quotes add a nice safety-check where if you forget the backslash, you get an error ("No closing quotation") instead of silent truncation.
> 
> According to [tox-dev/tox#2638](https://github.com/tox-dev/tox/issues/2638) just the backslash should be sufficient. I'll add it and switch back to a current tox version.

Ouch. tox 4 requires at least Python 3.7 so earlier Python versions will pull tox < 4 and choke on that backslash...

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2022-12-8 23:11 -0800:
>> Is there a chance just to disable the `isort` check (maybe via a configuration or a parameter)? It would be a pain to resolve merge conflicts caused by different sorting rules.
>
>Currently there is no such switch. But on the other hand there are currently no enforced sorting rules for this repository, so the possible conflicts have to be resolved only once.

But all merging conflicts would have to be resolved
and `isort`ing tends to provoke changes in many files
(not all causing merging conflicts).

**Comment by icemac** [comment]
> Ouch. tox 4 requires at least Python 3.7 so earlier Python versions will pull tox < 4 and choke on that backslash...

This hurts. Time to drop `py27-zodbmaster`? Or should I try to find a way to keep `py27-zodbmaster`?

**Comment by dataflake** [comment]
> > Ouch. tox 4 requires at least Python 3.7 so earlier Python versions will pull tox < 4 and choke on that backslash...
> 
> This hurts. Time to drop `py27-zodbmaster`? Or should I try to find a way to keep `py27-zodbmaster`?

IMHO the test itself can be dropped for GHA. I don't recall what the motivation was, maybe just an early warning for ZODB incompatibilities before a ZODB release? The added value may not be enough to justify the effort of keeping that alive.

File: .meta.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgedmin) [reply .meta.toml:17]: Oh, I see.  That makes sense, thank you for explaining!

**Review by dataflake — APPROVED** [review]

**Pull request merged**